descqa/CorrelationsTwoPoint.py

```diff
@@ -2,314 +2,175 @@
-            colors = plt.cm.jet(np.linspace(0, 1, len(filenames)))
-            mag_r_maxs = self.TestParams['mag_r_maxs']
-            mag_r_mins = self.TestParams['mag_r_mins']
-
-            #Filter on redshift space
-            zfilter = (lambda z: (z < self.TestParams['redshift_max']), 'redshift')
-            catalog_data_all = self.get_catalog_data(catalog_instance, ['ra', 'dec', mag_field], filters=[zfilter])
-            if catalog_data_all is None:
-                return TestResult(skipped=True, summary='Missing requested quantities')
-            plt.figure()
-            for rmax, rmin, fname, label, color in zip(mag_r_maxs, mag_r_mins, filenames, labels, colors):
-                validation_data = self.get_validation_data(self.observation, '{}_{}'.format(rmin, rmax))
-                catalog_data = GCRQuery((lambda mag: (mag > rmin) & (mag < rmax), mag_field)).filter(catalog_data_all)            
-                ra = catalog_data['ra']
-                dec = catalog_data['dec']
-
-                #It works only with rectangle area of sky
-                ramin, ramax = ra.min(), ra.max()
-                decmin, decmax = dec.min(), dec.max()
-                if ramin < 0:
-                    ra = np.mod(ra, 360)
-
-                del catalog_data
-
-                #plt.scatter(ra, dec)
-                #plt.savefig('s.png')
-
-                #Giving ra and dec to treecorr 
-                cat = treecorr.Catalog(ra=ra, dec=dec,  ra_units='degrees', dec_units='degrees')
-                dd = treecorr.NNCorrelation(min_sep=min_sep, max_sep=max_sep, bin_size=bin_size, sep_units='degrees')
+        # load catalog data
+        colnames = dict()
+        colnames['z'] = catalog_instance.first_available('redshift', 'redshift_true')
+        colnames['ra'] = catalog_instance.first_available('ra', 'ra_true')
+        colnames['dec'] = catalog_instance.first_available('dec', 'dec_true')
+        colnames['mag'] = catalog_instance.first_available(*self.possible_mag_fields)
+
+        if not all(v for v in colnames.values()):
+            return TestResult(skipped=True, summary='Missing requested quantities')
+
+        filters = [(np.isfinite, c) for c in colnames.values()]
+        filters.extend((
+            '{} < {}'.format(colnames['mag'], max(mag_bin['mag_max'] for mag_bin in self.mag_bins)),
+            '{} >= {}'.format(colnames['mag'], min(mag_bin['mag_min'] for mag_bin in self.mag_bins)),
+        ))
+        if self.need_distance:
+            filters.extend((
+                '{} < {}'.format(colnames['z'], max(mag_bin['cz_max'] for mag_bin in self.mag_bins)/self._C),
+                '{} >= {}'.format(colnames['z'], min(mag_bin['cz_min'] for mag_bin in self.mag_bins)/self._C),
+            ))
+        catalog_data = catalog_instance.get_quantities(list(colnames.values()), filters=filters)
+        catalog_data = {k: catalog_data[v] for k, v in colnames.items()}
+
+        # create random
```

Comment:
I think some times the sky area is slightly different for objects with different magnitude range. I think generating random catalog once is fine too.